### ISSUE TITLE
[Yaml] usage of conditional @deprecated-tag in Yaml::parse

### ISSUE THREAD

**Issue by gerritdrost** [symfony/symfony#14815]
Yaml::parse features a "conditional" `@deprecated` tag; the text after the tag explains that only the feature to supply filepaths is deprecated and will be removed in v3.0. PHPStorm however doesn't know this and marks the method as deprecated by striking through every call to it:
![yamlparse](https://cloud.githubusercontent.com/assets/5630275/7913892/8ab2f8da-0873-11e5-9a24-95738d091e47.png)

I know it is a good habit to mark stuff that will be removed as `@deprecated`, but I think in this case the `@deprecated` tag is a bit overkill since it also "deprecates" valid method calls. What are your thoughts on this?

**Comment by stof** [comment]
We should probably removed the ``@deprecated`` tag from places where the method is not deprecated but only some ways to use it. ``ExecutionContext->addViolation()`` has the same issue in the Validator component.

If anyone wants to work on this, it would be great.

**Comment by xabbuh** [comment]
The same applies to the `dumpFile()` method in the `Filesystem` class. Only using the `chmod` argument is deprecated.

**Comment by gerritdrost** [comment]
@xabbuh: I could not find a deprecated tag in `Filesystem::dumpFile` so it's not included in the pull request.

**Comment by gerritdrost** [comment]
@fabpot: Unfortunately it seems like something went wrong with the merge of this PR, the `Note:`-part was added but the `@deprecated`-tags weren't removed. I haven't had the time yet to check out the git logs yet though, so I can't tell you where exactly it went wrong. This does mean that this issue technically has not been resolved yet and should therefore probably be reopened.

**Comment by xabbuh** [comment]
see #14911

**Comment by gerritdrost** [comment]
Thanks! Could you also remove the `@deprecated` tag in `Yaml::parse`, since that also was missed out in the merge?

**Comment by xabbuh** [comment]
@gerritdrost That one looks good: https://github.com/symfony/symfony/blob/2.6/src/Symfony/Component/Yaml/Yaml.php#L25-52

**Comment by gerritdrost** [comment]
@xabbuh: Ah, I see. Problem is that in 2.7 the deprecated tags were not removed when my pull request was merged. This means that in 2.7 this issue is still valid, at least for Yaml, even after #14911 is merged. Do I create a new issue for that?

**Comment by jakzal** [comment]
@gerritdrost 2.6 was not merged to 2.7 yet.

**Comment by xabbuh** [comment]
@jakzal It seems it was, but adding only the note: 31eb16c03e40c902ea85a1808ff9d7ab311d818c Don't know what went wrong there.

**Comment by xabbuh** [comment]
@gerritdrost see #14925 for the final fix

**Comment by gerritdrost** [comment]
@xabbuh: awesome, thanks!
